File: Count_Negative/count.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public int CountNegatives(int[][] grid)
    {
        int result = 0;

        for (int i = 0; i < grid.Length; i++)
        {
            result += binarySearch(grid[i]);
        }

        return result;
    }

    int binarySearch(int[] target)
    {
        int i = 0,
            j = target.Length - 1;

        while (i <= j)
        {
            int mid = (int)(i + j) / 2;
            if (target[mid] >= 0)
            {
                i = mid + 1;
            }
            else
            {
                j = mid - 1;
            }
        }

        int count = 0;

        for (int k = i; k < target.Length; k++)
        {
            count++;
        }

        return count;
    }
}

File: Count_Negative/count.go
package countnegative

unc countNegatives(grid [][]int) int {
	result := 0

	for _, arr := range grid {
		result += binarySearch(arr)
	}

	return result
}

func binarySearch(target []int) int {
	i, j := 0, len(target)-1

	for i <= j {
		mid := int((i + j) / 2)

		if target[mid] >= 0 {
			i = mid + 1
		} else {
			j = mid - 1
		}
	}

	return len(target[i:])
}
